src/main/java/team/reborn/energy/impl/EnergyImpl.java

```diff
@@ -51,4 +56,23 @@ public long getCapacity() {
 			return 0;
 		}
 	};
+
+	public static final DataComponentType<Long> ENERGY_COMPONENT = Registry.register(
+			Registries.DATA_COMPONENT_TYPE,
+			new Identifier("team_reborn_energy", "energy"),
+			DataComponentType.<Long>builder()
+					.codec(nonNegativeLong())
+					.packetCodec(PacketCodecs.VAR_LONG)
+					.build()
+	);
+
+	private static Codec<Long> nonNegativeLong() {
+		return Codecs.validate(Codec.LONG, (Long value) -> {
+			if (value.compareTo(0L) >= 0 && value.compareTo(Long.MAX_VALUE) <= 0) {
```

Comment:
Isn't the second always true?

Reply:
I copied this from vanilla, ill double check and add a test.